hyperlane/relayer_runner.go

```diff
@@ -83,8 +83,46 @@ func (r *RelayerRunner) Run(ctx context.Context) error {
 					continue
 				}
 
+				isTransferValid, err := r.validateHyperlaneTransfer(ctx, transfer)
+				if err != nil {
+					lmt.Logger(ctx).Warn(
+						"failed to validate Hyperlane transfer",
+						zap.Error(err),
+						zap.Int64("transferId", transfer.ID),
+						zap.String("txHash", transfer.MessageSentTx),
+					)
+					continue
+				}
+
+				if !isTransferValid {
+					lmt.Logger(ctx).Warn(
+						"abandoning invalid hyperlane transfer",
+						zap.Any("transfer", transfer),
+						zap.Error(err),
+					)
+
+					if _, err := r.db.SetMessageStatus(ctx, db.SetMessageStatusParams{
+						TransferStatus:     dbtypes.TransferStatusAbandoned,
+						SourceChainID:      transfer.SourceChainID,
+						DestinationChainID: transfer.DestinationChainID,
+						MessageID:          transfer.MessageID,
+					}); err != nil {
+						lmt.Logger(ctx).Error(
+							"error updating invalid transfer status",
+							zap.Error(err),
+							zap.Int64("transferId", transfer.ID),
+							zap.String("txHash", transfer.MessageSentTx),
+						)
+					}
+					continue
+				}
+
 				destinationTxHash, destinationChainID, err := r.relayHandler.Relay(ctx, transfer.SourceChainID, transfer.MessageSentTx)
 				if err != nil {
+					if errors.Is(err, ErrNotEnoughSignaturesFound) {
+						// warning already logged in relayer
```

Comment:
this is to prevent spam of 

`{"level":"error","ts":1731035285.0626438,"caller":"hyperlane/relayer_runner.go:88","msg":"error relaying pending hyperlane transfer","error":"getting checkpoint at index 328: expected atleast 3 signatures in multisig signed checkpoint, but got 0","sourceChainID":"osmosis-1","txHash":"9C4EDA7213A3909E7F852B4047BFE74F76CD2477C8C92CA1975DD70B74547F73","stacktrace":"[github.com/skip-mev/go-fast-solver/hyperlane.(*RelayerRunner).Run](http://github.com/skip-mev/go-fast-solver/hyperlane.(*RelayerRunner).Run)\n\t/solver/hyperlane/relayer_runner.go:88\nmain.main.func6\n\t/solver/cmd/solver/main.go:153\[ngolang.org/x/sync/errgroup.(*Group).Go.func1](http://ngolang.org/x/sync/errgroup.(*Group).Go.func1)\n\t/root/.cache/go-build/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:78"}`